File: source_funcs/go_transcoder/MINIMUM_NUMBER_PLATFORMS_REQUIRED_RAILWAYBUS_STATION/MINIMUM_NUMBER_PLATFORMS_REQUIRED_RAILWAYBUS_STATION.go
package main

import "sort"

func f_gold(arr []int, dep []int, n int) int {
	sort.Ints(arr)
	sort.Ints(dep)
	var plat_needed int = 1
	var result int = 1
	var i int = 1
	var j int = 0
	for i < n && j < n {
		if arr[i] <= dep[j] {
			plat_needed++
			i++
			if plat_needed > result {
				result = plat_needed
			}
		} else {
			plat_needed--
			j++
		}
	}
	return result
}

File: rust_funcs/c_transcoder/MINIMUM_NUMBER_PLATFORMS_REQUIRED_RAILWAYBUS_STATION/MINIMUM_NUMBER_PLATFORMS_REQUIRED_RAILWAYBUS_STATION.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::cmp::Ordering;

fn f_gold(arr: [i32;2], dep: [i32;2], n: i32) -> i32 {
    let mut arr = arr.to_vec();
    let mut dep = dep.to_vec();
    arr.sort();
    dep.sort();
    let mut plat_needed = 1;
    let mut result = 1;
    let mut i = 1;
    let mut j = 0;
    while i < n as usize && j < n as usize {
        match arr[i].cmp(&dep[j]) {
            Ordering::Less => {
                plat_needed += 1;
                i += 1;
                result = std::cmp::max(result, plat_needed);
            }
            Ordering::Greater => {
                plat_needed -= 1;
                j += 1;
            }
            Ordering::Equal => {
                i += 1;
                j += 1;
            }
        }
    }
    result
}
